anomalous-value-js/src/agent.spec.js

```diff
@@ -0,0 +1,128 @@
+  const testAddress = '0x1234';
+  let contractAddresses = {};
+  contractAddresses[testAddress] = mockRollingMath;
+
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(contractAddresses, provider);
+  });
+
+  describe("handleTransaction", () => {
+
+    it("returns empty findings if there is no 'to' field, as in contract creation", async () => {
+      const txEvent = createTxEvent(null, 10);
+      const findings = await handleTransaction(txEvent);
+      
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns empty findings and adds to contractAddresses if this is the first time seeing a contract", async() => {
+      const testContractAddress = '0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9' //Aave token address, need a random contract not in 'contractAddresses'
+      
+      const txEvent = createTxEvent(testContractAddress, 10);
+      const findings = await handleTransaction(txEvent);
+      
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns empty findings if 'to' field is an EOA", async() => {
+      const testEOA = '0x4683e61663D1dF94340D09E3Ed92D7B05aF2FdB0';
+
+      const txEvent = createTxEvent(testEOA, 10);
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns empty findings if the value field is within tolerance", async() => {
+      mockRollingMath.getAverage.mockReturnValueOnce(gwei(10));
+      mockRollingMath.getStandardDeviation.mockReturnValueOnce(gwei(1));
+      mockRollingMath.getNumElements.mockReturnValueOnce(41);
+
+      const value = gwei(11);
+      const txEvent = createTxEvent(testAddress, value);
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(mockRollingMath.getAverage).toHaveBeenCalledTimes(1);
+      expect(mockRollingMath.getStandardDeviation).toHaveBeenCalledTimes(1);
+      expect(mockRollingMath.getNumElements).toHaveBeenCalledTimes(0);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns a finding if the value field is outside tolerance", async() => {
+      mockRollingMath.getAverage.mockReset();
+      mockRollingMath.getStandardDeviation.mockReset(); 
+      mockRollingMath.getNumElements.mockReset(); 
+
+      mockRollingMath.getAverage.mockReturnValueOnce(gwei(10));
+      mockRollingMath.getStandardDeviation.mockReturnValueOnce(gwei(1));
+      mockRollingMath.getNumElements.mockReturnValueOnce(41);
+
+      const value = gwei(20);
+      const txEvent = createTxEvent(testAddress, value);
+
+      const findings = await handleTransaction(txEvent);
+
+      expect(mockRollingMath.getAverage).toHaveBeenCalledTimes(1);
+      expect(mockRollingMath.getStandardDeviation).toHaveBeenCalledTimes(1);
+      expect(mockRollingMath.getNumElements).toHaveBeenCalledTimes(1);
+      
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "High Value",
+          description: `Value: ${value}`,
```

Comment:
Might want to include the contract address here as well:
"Transaction with contract 0xABCD had unusually high value of XYZ Ether"